doc/error-handling/SONiC_Mgmt_Error_Framework_HLD.md

```diff
@@ -0,0 +1,241 @@
+https://github.com/Azure/SONiC/blob/b8816dfac04b17e0e7bc7a68361e4feab1ae47c5/doc/error-handling/error_handling_design_spec.md
+
+# Definition/Abbreviation
+
+### Table 1: Abbreviations
+| **Term**                 | **Meaning**                         |
+|--------------------------|-------------------------------------|
+| NBI                      | North Bound Interface               |
+
+# 1 Feature Overview
+
+The feature addresses the north-bound interfaces (CLI/REST/gNMI) for configuring  and fetching the data from the back-end to the user using SONiC managment framework. The  implementation is contained within the Management Framework container.
+
+Their is no openconfig data yang defined for this feature. A new SONiC yang will be introduced for the NBI.
+
+The following operations will be exposed by the SONiC yang interface:
+1. Get errors from redis:ErrorDB.
+2. Clear error tables in redis:ErrorDB.
+
+## 1.1 Requirements
+The requirements for the feature are derived from the Error Handling Framework Specification.
+https://github.com/Azure/SONiC/blob/b8816dfac04b17e0e7bc7a68361e4feab1ae47c5/doc/error-handling/error_handling_design_spec.md
+
+### 1.1.1 Functional Requirements
+Provide yang interface for REST and CLI commands for feature configuration and management.
+
+### 1.1.2 Configuration and Management Requirements
+
+- CLI commands
+
+Mode: Show
+1. show error-database \[*tablename*\] <br>
+  Parameter: *tablename*: ERROR_NEIGH_TABLE, ERROR_ROUTE_TABLE <br>
+
+Mode: Configuration
+1. no error-database \[*table-name*\] <br>
+  Parameter: *tablename*: ERROR_NEIGH_TABLE, ERROR_ROUTE_TABLE <br>
+
+
+REST support
+gNMI support for show errors tables.
+
+
+### 1.1.3 Scalability Requirements
+Not Applicable
+### 1.1.4 Warm Boot Requirements
+Not Applicable
+
+## 1.2 Design Overview
+### 1.2.1 Basic Approach
+The 'show' command will fetch the error table information from ErrorDB(8). This is through yang show construct.
+The 'clear error-database [*tablename*]' is handle through yang rpc construct. It provides the mechanism to register a callback. The callback will send notification to ErrorDB subscribers, which in response will clear the tables.
+
+### 1.2.2 Container
+This feature is contained within the sonic-mgmt-framework container.
+
+### 1.2.3 SAI Overview
+Not Applicable.
+
+# 2 Functionality
+## 2.1 Target Deployment Use Cases
+Not applicable
+## 2.2 Functional Description
+Wordy description
+
+# 3 Design
+## 3.1 Overview
+Not applicable.
+## 3.2 DB Changes
+Not applicable.
+## 3.3 Switch State Service Design
+Not applicable.
+
+## 3.4 SyncD
+Not applicable.
+
+## 3.5 SAI
+Not applicable.
+
+## 3.6 User Interface
+### 3.6.1 Data Models
+
+A new sonic yang (sonic-error.yang) provides the interface for configuration and status.
+```
+module: sonic-error
+    +--rw sonic-error
+       +--rw ERROR_NEIGH_TABLE
+       |  +--rw ERROR_NEIGH_TABLE_LIST* [ifname prefix]
```

Comment:
Key in the neighbor table is interface and IP address. Please replace prefix with IP address